lib/kafka/compression.rb

```diff
@@ -0,0 +1,23 @@
+require "kafka/snappy_codec"
+require "kafka/gzip_codec"
+
+module Kafka
+  module Compression
+    def self.find_codec(name)
+      case name
+      when nil then nil
+      when :snappy then SnappyCodec.new
+      when :gzip then GzipCodec.new
```

Comment:
Probably smarter, but definitely not easier to read. This is dumb and readable, and works great.